modules/es-cluster-health-is-red.adoc

```diff
@@ -0,0 +1,153 @@
+// Module included in the following assemblies:
+//
+// * logging/logging_alerts/troubleshooting-logging-alerts.adoc
+
+:_content-type: PROCEDURE
+[id="es-cluster-health-is-red_{context}"]
+= Elasticsearch cluster health status is red
+
+At least one primary shard and its replicas are not allocated to a node. Use the following procedure to troubleshoot this alert.
+
+.Procedure
+
+. Check the Elasticsearch cluster health and verify that the cluster `status` is red by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- health
+----
+
+. List the nodes that have joined the cluster by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/nodes?v
+----
+
+. List the Elasticsearch pods and compare them with the nodes in the command output from the previous step, by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-logging get pods -l component=elasticsearch
+----
+
+. If some of the Elasticsearch nodes have not joined the cluster, perform the following steps.
+
+.. Confirm that Elasticsearch has an elected control plane node by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/master?v
+----
+
+.. Review the pod logs of the elected control plane node for issues by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc logs <elasticsearch_master_pod_name> -c elasticsearch -n openshift-logging
+----
+
+.. Review the logs of nodes that have not joined the cluster for issues by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc logs <elasticsearch_node_name> -c elasticsearch -n openshift-logging
+----
+
+. If all the nodes have joined the cluster, check if the cluster is in the process of recovering by running the following command and observing the output:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/recovery?active_only=true
```

Comment:
```suggestion
$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query="_cat/recovery?active_only=true"
```

Reply:
Per https://redhat-internal.slack.com/archives/C02321AKYTT/p1696866637850469?thread_ts=1696524477.252959&cid=C02321AKYTT we will leave this as is